lib/ansible/plugins/lookup/file_found.py

```diff
@@ -0,0 +1,123 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: file_found
+    author: Pierre Monsimier <pierre.monsimier@capgemini.com>
+    short_description: return first file found from list
+    description:
+      - this lookup checks a list of files and paths and returns a list of full paths of those found.
+      - As all lookups, when fed relative paths it will try use the current task's location first and go up the chain
+        to the containing role/play/include/etc's location.
+      - The list of files has precedence over the paths searched.
+        i.e, A task in a  role has a 'file1' in the play's relative path, this will be used, 'file2' in role's relative path will not.
+    options:
+      _terms:
+        description: list of file names
+        required: True
+      paths:
+        description: list of paths in which to look for the files
```

Comment:
doesn't this overlap a lot with fileglob/filetree> ... just like using w/o the *